### PR TITLE
Update into Intro.mdx

### PR DESCRIPTION
**Pull request by amjedidiah**
More explanation on implementation details to avoid testing for

### CODE REVIEW

File: docs/intro.mdx [hunk 1]
```diff
@@ -54,7 +54,7 @@ You may want to avoid the following implementation details:
 1. Internal state of a component
 1. Internal methods of a component
 1. Lifecycle methods of a component
-1. Child components
+1. Child components, i.e: each component(parent and children) should have its own test file
```
Comment (ph-fritsche) [inline docs/intro.mdx:57]: This is the opposite of what is encouraged in this paragraph.

Testing Library encourages you to test your components through integration. A child component is an implementation detail.

Comment (MatanBobi) [reply docs/intro.mdx:57]: Integration tests aims to test a use-case or a user-flow, so in this case I'd suggest testing the `Login` component :)

Comment (amjedidiah) [reply docs/intro.mdx:57]: Alright then.
And should the form component still be used in a Register component, the test will be written for the Register component and not the form component, yes?

Comment (MatanBobi) [reply docs/intro.mdx:57]: Yeah, exactly.
If you have a `Form` component which is a generic one and you use it multiple time in different cases, you might want to also unit test that component. I recommend reading this one:
https://matan.io/posts/to-unit-or-integration

**Comment by MatanBobi** [comment]
Closing as this change isn't what we aim to push :)
Thanks for the effort!

File: docs/intro.mdx (hunk 1 above)
Comment (amjedidiah) [reply docs/intro.mdx:57]: Alright, thanks man.
Better understood.

**Pull request closed without merging**